lock-impl/src/main/java/com/palantir/lock/impl/LegacyTimelockService.java

```diff
@@ -183,14 +185,14 @@ public long currentTimeMillis() {
     }
 
     @Override
-    public AcquireNamedMinTimestampLeaseResult acquireNamedMinTimestampLease(
-            String timestampName, int numFreshTimestamps) {
+    public Map<TimestampLeaseName, TimestampLeaseResult> acquireTimestampLeases(
```

Comment:
Word of caution that these might be tricky to implement here, I had to understand how the rest of this class relies on custom immutableTsLockClient instances to get the immutable lock implementation. I think you can make this work and we NEED this before we're able to use this in an internal queueing library, because otherwise tests won't pass.

I would LOVE to get rid of this implementation and just pull in an in-memory timelock for tests, because it's super annoying to have to have this implementation and impl everything twice